tensorflow_similarity/indexer.py

```diff
@@ -135,31 +119,30 @@ def _init_structures(self) -> None:
         "(re)initialize internal storage structure"
 
         if self.search_type == "nmslib":
-            self.search: Search = NMSLibSearch(distance=self.distance, dim=self.embedding_size)
-        elif isinstance(self.search_type, Search):
-            self.search = self.search_type
-        else:
+            self.search = NMSLibSearch(distance=self.distance, dim=self.embedding_size)
+        elif self.search_type == "linear":
+            self.search = LinearSearch(distance=self.distance, dim=self.embedding_size)
+        elif not isinstance(self.search, Search):
+            # self.search should have been already initialized
             raise ValueError("You need to either supply a known search " "framework name or a Search() object")
 
         # mapper from id to record data
         if self.kv_store_type == "memory":
-            self.kv_store: Store = MemoryStore()
+            self.kv_store = MemoryStore()
         elif isinstance(self.kv_store_type, Store):
```

Comment:
I feel having kv_store_type be an object is kinda against the spirit of its variable name.
We basically have 3 pieces of information:
- the class (needs to be instantiated)
- the shorthand name for class which is kinda arbitrary ("memory" for "MemoryStore")
- and instantiated instance of the class
What complicates things is the fact that we need to be able to reinstantiate these objects in the reset function, which I'm not sure how to do if the given parameter was an instantiated object and not a type (since all the parameters for its instantiation will be hidden from the init method).
If we need reset capability I feel like we need to not support objects, but only classes, and always instantiate them inside the init.
what do you think?

Reply:
Oh, sorry, I should have worded my comment better. Here I meant we are checking if self.kv_store_type is a Store instance, but I don't think that's possible (to your points above).

Reply:
done

Reply:
got it, I think that case is unreachable, so removing it